### PR TITLE
DAOS-10683 build: Restrict SPDK version to less than 22

### PR DESCRIPTION
**Pull request by tanabarr**
Restrict SPDK version on release/2.0 to avoid any interference when
upgrading to v22.01.1 on master branch.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by NiuYawei — DISMISSED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Basically all of the same concerns expressed in #9057.

File: debian/control [hunk 1]
```diff
@@ -21,12 +21,12 @@ Build-Depends: debhelper (>= 10),
                libfuse3-dev,
                libprotobuf-c-dev,
                libjson-c-dev,
-               dpdk-dev,
+               dpdk-dev (<<21.11),
```
Comment (brianjmurrell) [inline debian/control:24]: Ahh.  Interesting that you spotted this.  DAOS doesn't have a direct dependency on DPDK right?  That's what the `daos.spec` is expressing at least.  This should likely just be removed from this file:

```suggestion
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
You are missing the `debian/changes` update in this PR.

File: debian/control (hunk 1 above)
Comment (tanabarr) [reply debian/control]: done

Comment (tanabarr) [reply debian/control:24]: this change is causing build failures, reverting

**Review by NiuYawei — APPROVED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by kjacque — APPROVED** [review]

**Comment by brianjmurrell** [comment]
@tanabarr Can you limit your changes to a single branch's version of this PR until everything is worked out only once it's fully working on that branch port to the others?  I am getting overload and confused on reviewing each change 3 times.  Or maybe make your other 2 PRs drafts so that reviewers are not constantly updated on each incremental addtion?

**Comment by tanabarr** [comment]
> @tanabarr Can you limit your changes to a single branch's version of this PR until everything is worked out only once it's fully working on that branch port to the others? I am getting overload and confused on reviewing each change 3 times. Or maybe make your other 2 PRs drafts so that reviewers are not constantly updated on each incremental addtion?

I've marked this one as draft and will just work on the release/2.2 version until it's passing. I am not sure which other PR you're talking about when you mention "other 2 PRs" as the only other pin PR is https://github.com/daos-stack/daos/pull/9057 .

**Review by kjacque — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**